src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/base/Functions.java

```diff
@@ -63,6 +63,17 @@
      * function that returns the length of an array
```

Comment:
This is an excellent candidate location for the function  catalogue. We have only names here, but you simply need to convert these static constants to a (possibly static as well) map from name+arity to signature + pointers to runtime iterator classes.